javascript/javascript1/week2/student-manager.js

```diff
@@ -0,0 +1,62 @@
+const class07Students = [];
+function addStudentToClass(studentName) {
+    if (studentName === '' || studentName === undefined || studentName === null) {
+        return console.log(`Please provide a valid student name`);
+    }
+    else if (!istudentExist(studentName)) {
+        if (class07Students.length < 6 || studentName.toUpperCase() === 'QUEEN') {
+            console.log(`${studentName} has been added successfully`);
+            class07Students.push(studentName);
+        }
+        else {
+            console.log(`Sorry the class is full to enroll for any more student`);
+        }
+    }
+    else
+        console.log(`${studentName} is already exists in the class`);
+}
+
+let istudentExist = studentName => class07Students.find(name => name.toLowerCase() === studentName.toLowerCase()
+) === undefined ? false : true;
+
+//traditional function definition for the above arrow function
+// function istudentExist(studentName) {
+//     let outPut = class07Students.find(nameCheck);
+//     function nameCheck(name) {
+
+//         if (name.toLowerCase() === studentName.toLowerCase()) {
+//             return studentName;
+//         }
+//     }
+//     if (outPut === undefined) {
+//         return false;
+//     }
+//     else {
+//         return true;
+//     }
+// }
+
+const getNoOfStudents = class07Students => console.log(`The number of total students in the class is  ${class07Students.length}`);
```

Comment:
this variable returns zero. I will leave that to you to find out because it is quite important to understand why.

Reply:
Hi
I just tried to keep this an unsued variable to ternminate the loop,my motivation was here to terminate the loop once the total price is less than amount to spend and till that to continue the loop and so I used a flag variable and and once its false terminated the loop,so just for false value used adummy value 0.